### DISCUSSION TITLE
CPU overload after migrating to v2.1.0

### DISCUSSION THREAD

**Discussion by Tony-MF** [plausible/analytics#4128] [Self-Hosted Support] [answered]
Hi,

I have an issue with the new version.

If I add these lines to my nginx confix:

_proxy_http_version 1.1;
proxy_set_header Upgrade $http_upgrade;
proxy_set_header Connection "Upgrade";_

it increases the system load from about 7 to 46. Here is the CPU utilization after I remove those lines: 
<img width="1165" alt="image" src="https://github.com/plausible/analytics/assets/73653258/987c9008-935c-416c-ae8c-189c76a667e7">

For now I removed them, but then the spark charts don't load. 

This is my nginx conf:

```
        location / {
            proxy_pass http://127.0.0.1:8000;

            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
#           proxy_http_version 1.1;
#           proxy_set_header Upgrade $http_upgrade;
#           proxy_set_header Connection "Upgrade";
            proxy_set_header Host $host;
            proxy_set_header X-Real-IP $remote_addr;
        }
```

**Comment by ruslandoga** [comment]
👋 @Tony-MF 

What process is in purple, is it Plausible or Nginx or something else?

**Reply by ruslandoga** [reply]
Also was there anything interesting in `docker compose logs` during the overload period?

**Comment by mebeim** [comment, marked as answer]
Don't know if this is what is causing the issue, but I think that the nginx config in the community-edition repo is ill-formed. It is currently suggesting to add those settings globally (i.e. under `location /` so for every location), but that is not how it should be done. Only the websocket endpoint(s) should have those special settings, otherwise you will set `Connection: upgrade` for any connection and also use HTTP 1.1 when potentially a higher version could be used.

In my instance, I use something like this instead:

```none
	location / {
		proxy_pass http://127.0.0.1:8000;
		// ...

		location = /live/websocket {
			proxy_pass         http://127.0.0.1:8000;
			proxy_http_version 1.1;
			proxy_set_header   Upgrade    $http_upgrade;
			proxy_set_header   Connection "upgrade";
		}
	}
```

See also this thread https://trac.nginx.org/nginx/ticket/2115 where Nginx maintainers say more or less the same.

**Reply by Tony-MF** [reply]
Yes, that was it! Thanks a lot @mebeim 🚀

**Reply by ruslandoga** [reply]
@mebeim 👋 

PR is welcome!

**Reply by adrianjean** [reply]
This helped me as well. Thank you!
